crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -264,6 +264,13 @@ impl SnapshotProvider {
                             index.insert(tx_end, current_block_range.clone());
                         })
                         .or_insert_with(|| BTreeMap::from([(tx_end, current_block_range)]));
+                } else if let Some(1) = tx_index.get(&segment).map(|index| index.len()) {
+                    // Only happens if we unwind all the txs/receipts from the first static file.
+                    // Should only happen in test scenarios.
```

Comment:
hmm, isn't it possible that we have index of length one during the normal node operation?

Reply:
i may be missing something, but i think highly unlikely...